### PR TITLE
Baselines the MAINTAINERS.md and CODEOWNERS files.

### LINKED ISSUE #196: Baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions
Follow https://github.com/opensearch-project/.github/issues/125 to baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions.

Close this issue when:

- [ ] 1. [MAINTAINERS.md](MAINTAINERS.md) has the correct list of project maintainers.
- [ ] 2. [CODEOWNERS](CODEOWNERS) exists and has the correct list of aliases.
- [ ] 3. Repo permissions only contain individual aliases as collaborators with maintain rights, admin, and triage teams. 
- [ ] 4. All other teams are removed from repo permissions.

If this repo's permissions was already baselined, please confirm the above when closing this issue.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description
Baselines the MAINTAINERS.md and CODEOWNERS files.

@VijayanB , @VachaShah 
@asifsmohammed , @oeyh 

### Issues Resolved
Resolves #196

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by deepdatta — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]

**Comment by dlvenable** [comment]
@opensearch-project/admin , Please update the permissions on this project to match the users in the CODEOWNERS file in this PR.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,4 @@
-# This should match the owning team set up in https://github.com/orgs/opensearch-project/teams
-*   @opensearch-project/logstash-output-opensearch
\ No newline at end of file
+# This should match the list of maintainers in MAINTAINERS.md
+* @asifsmohammed
+* @dlvenable
+* @oeyh
```
[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/CODEOWNERS:4]: Aah! That's why @dlvenable won here https://github.com/opensearch-project/spring-data-opensearch/blob/main/.github/CODEOWNERS in my [PR](https://github.com/opensearch-project/spring-data-opensearch/pull/101#issuecomment-1456766267)

**Review by peternied — COMMENTED** [review]
I've updated the permissions in the settings, let me know if this should be altered

image

File: .github/CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlvenable) [reply .github/CODEOWNERS:4]: Thanks for the tip. Keeping all on a single line is more cumbersome, but it seems it is what we need to do. I went ahead and merged and raised #205 to fix this.

**Pull request merged**